drivers/sx126x/sx126x.c

```diff
@@ -190,20 +192,83 @@ int sx126x_init(sx126x_t *dev)
     sx126x_init_default_config(dev);
 
     /* Configure available IRQs */
-    const uint16_t irq_mask = (
-        SX126X_IRQ_TX_DONE |
-        SX126X_IRQ_RX_DONE |
-        SX126X_IRQ_PREAMBLE_DETECTED |
-        SX126X_IRQ_SYNC_WORD_VALID |
-        SX126X_IRQ_HEADER_VALID |
-        SX126X_IRQ_HEADER_ERROR |
-        SX126X_IRQ_CRC_ERROR |
-        SX126X_IRQ_CAD_DONE |
-        SX126X_IRQ_CAD_DETECTED |
-        SX126X_IRQ_TIMEOUT
-        );
+    uint16_t irq_mask_dio1 = dev->params->dio1_irq_mask;
+    uint16_t irq_mask_dio2 = 0;
+    uint16_t irq_mask_dio3 = 0;
+
+#if IS_USED(MODULE_SX126X_DIO2)
+    if (dev->params->dio2_mode == SX126X_DIO2_IRQ) {
+        if (gpio_is_valid(dev->params->u_dio2_arg.dio2_pin)) {
+            res = gpio_init_int(dev->params->u_dio2_arg.dio2_pin, GPIO_IN, GPIO_RISING,
+                                dev->event_cb, dev->event_arg);
+            if (res < 0) {
+                DEBUG("[sx126x] error: failed to initialize DIO2 pin\n");
+                return res;
+            }
+            irq_mask_dio2 = dev->params->u_dio2_arg.dio2_irq_mask;
+        }
+    }
+    else if (dev->params->dio2_mode == SX126X_DIO2_RF_SWITCH) {
+        if (gpio_is_valid(dev->params->u_dio2_arg.rf_switch_pin)) {
```

Comment:
typical use of rf switch is not another  pin but actually a external rf switch (antenna) that is switched by this

Reply:
This is actually what we have. Technically if it was connected to an MCU pin, it could be set up with an interrupt. 
I am going to remove `rf_switch_pin`.